src/displayBackend/wayland/Display.cpp

```diff
@@ -554,6 +568,12 @@ void Display::release()
 #endif
 #endif
 
+#ifdef WITH_WAYLAND_PRESENTATION_API
+	if(mCompositor)
+		mCompositor->setPresentation(nullptr);
+	wp_presentation_destroy(mPresentation);
```

Comment:
presentation implementation is shared between different classes. I mean different pieces of presentation is implemented in Display, Compositor, Surface etc. I suggest to implement new class Presentation: `class Presentation : public Registry` bind it in the constructor, destroy it in the destructor. Add method to wrap add feedback listener. In display create shared pointer to `Presentation` pass it to `Compositor` and then to `Surface`.

Reply:
Ok . I will try.